VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/VitruvioMesh.cpp

```diff
@@ -0,0 +1,100 @@
+#include "VitruvioMesh.h"
+
+#include "MaterialConversion.h"
+#include "StaticMeshAttributes.h"
+
+UMaterialInstanceDynamic* CacheMaterial(UMaterial* OpaqueParent, UMaterial* MaskedParent, UMaterial* TranslucentParent,
+										TMap<FString, Vitruvio::FTextureData>& TextureCache,
+										TMap<Vitruvio::FMaterialAttributeContainer, UMaterialInstanceDynamic*>& MaterialCache,
+										const Vitruvio::FMaterialAttributeContainer& MaterialAttributes, const FName& Name, UObject* Outer)
+{
+	if (MaterialCache.Contains(MaterialAttributes))

```

Comment:
AMaybe you want to work with an iterator to avoid duplicate lookup?

Reply:
Good point @mistafunk! Will do.